mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/ConvertScreenV2.tsx

```diff
@@ -0,0 +1,355 @@
+  const hasPendingBroadcastJob = useSelector((state: RootState) => hasBroadcastQueued(state.ocean))
+  const navigation = useNavigation<NavigationProp<PortfolioParamList>>()
+  const [mode, setMode] = useState(props.route.params.mode)
+  const [sourceToken, setSourceToken] = useState<ConversionIO>()
+  const [targetToken, setTargetToken] = useState<ConversionIO>()
+  const [convAmount, setConvAmount] = useState<string>('0')
+  const [fee, setFee] = useState<BigNumber>(new BigNumber(0.0001))
+  const [amount, setAmount] = useState<string>('')
+  const [hasError, setHasError] = useState<boolean>(false)
+  const [showMaxUTXOWarning, setShowMaxUTXOWarning] = useState<boolean>(false)
+  const [transactionCardStatus, setTransactionCardStatus] = useState<'error' | 'active' | undefined>()
+  const [isInputFocus, setIsInputFocus] = useState<boolean>(false)
+
+  useEffect(() => {
+    client.fee.estimate()
+      .then((f) => setFee(new BigNumber(f)))
+      .catch(logger.error)
+  }, [])
+
+  useEffect(() => {
+    const [source, target] = getDFIBalances(mode, tokens)
+    setSourceToken(source)
+    setTargetToken(target)
+    const sourceNum = new BigNumber(source?.amount !== undefined && source.amount !== '' ? source.amount : 0)
+    const conversionNum = new BigNumber(amount).isNaN() ? new BigNumber(0) : new BigNumber(amount)
+    const conversion = conversionNum.toString()
+    setConvAmount(conversion)
+    setHasError(conversionNum.gt(sourceNum))
+    setShowMaxUTXOWarning(isUtxoToAccount(mode) && !sourceNum.isZero() && conversionNum.toFixed(8) === sourceNum.toFixed(8))
+  }, [mode, JSON.stringify(tokens), amount])
+
+  useEffect(() => {
+    setTransactionCardStatus(hasError ? 'error' : isInputFocus ? 'active' : undefined)
+  }, [hasError, isInputFocus])
+
+  if (sourceToken === undefined || targetToken === undefined) {
+    return <></>
+  }
+
+  function convert (sourceToken: ConversionIO, targetToken: ConversionIO): void {
+    if (hasPendingJob || hasPendingBroadcastJob) {
+      return
+    }
+    navigation.navigate({
+      name: 'ConvertConfirmationScreenV2',
+      params: {
+        sourceUnit: sourceToken.unit,
+        sourceBalance: BigNumber.maximum(new BigNumber(sourceToken.amount).minus(convAmount), 0),
+        targetUnit: targetToken.unit,
+        targetBalance: BigNumber.maximum(new BigNumber(targetToken.amount).plus(convAmount), 0),
+        mode,
+        amount: new BigNumber(amount),
+        fee
+      },
+      merge: true
+    })
+  }
+
+  function onPercentagePress (amount: string, type: AmountButtonTypes): void {
+    setAmount(amount)
+    showToast(type)
+  }
+
+  function showToast (type: AmountButtonTypes): void {
+    toast.hideAll()
+    const isMax = type === AmountButtonTypes.max
+    const toastMessage = isMax ? 'Max available {{unit}} entered' : '{{percent}} of available {{unit}} entered'
+    const toastOption = {
+      unit: getDisplayUnit(sourceToken?.unit),
```

Comment:
we can do an early `return` if `sourceToken` is undefined, so `getDisplayUnit()` doesn't have to accept nullable param

Reply:
done, added `return` on `sourceToken` undefined